src/core/wee-crypto.c

```diff
@@ -524,6 +524,10 @@ weecrypto_totp_generate_internal (const char *secret, int length_secret,
     int rc, offset, length;
     unsigned long bin_code;
 
+#if __BYTE_ORDER == __BIG_ENDIAN
```

Comment:
Hi @luk1337,
I just noticed that the add of this condition make tests fail and procude invalid TOTP on FreeBSD (I'm testing with FreeBSD 14.0 on KVM/QMU with x86_64 arch).
If this condition is really needed on s390x, could it be changed to be more restrictive and apply only on this OS?

Reply:
https://github.com/weechat/weechat/pull/2171, idk if it will compile on all platforms tho, let's see.